### ISSUE TITLE
Panic when improperly ignoring stainless test

### ISSUE THREAD

**Issue by austinglaser** [rust-lang/rust#43160]
I'm new to using the stainless plugin for my tests. While writing them, I ignored one the wrong way (using the `#[ignore]` annotation), and caused a rustc panic.

I reduced my code to a MWE:

I tried this code:

**Cargo.toml:**

```
[package]
name = "rustc-panic"
version = "0.1.0"
authors = ["Austin Glaser <austin.glaser@gmail.com>"]

[dependencies]

[dev-dependencies]
stainless = "*"
```

**src/lib.rs:**

```
#![feature(plugin)]
#![cfg_attr(test, plugin(stainless))]

#[cfg(test)]
describe! a_thing {
    #[ignore]
    it "should be ignored" {
        assert!(false);
    }   
}
```

I expected to see this happen: A compiler error, complaining about my bogus syntax

Instead, this happened:

```
$ RUST_BACKTRACE=1 cargo test --verbose
       Fresh stainless v0.1.11
   Compiling rustc-panic v0.1.0 (file:///home/austin/scratch/rustc-panic)
     Running `rustc --crate-name rustc_panic src/lib.rs --emit=dep-info,link -C debuginfo=2 --test -C metadata=86f20b59323deae2 -C extra-filename=-86f20b59323deae2 --out-dir /home/austin/scratch/rustc-panic/target/debug/deps -L dependency=/home/austin/scratch/rustc-panic/target/debug/deps --extern stainless=/home/austin/scratch/rustc-panic/target/debug/deps/libstainless-6e9196cf7cd4f8c1.so`
error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.20.0-nightly (bf0a9e0b4 2017-07-10) running on x86_64-unknown-linux-gnu

note: run with `RUST_BACKTRACE=1` for a backtrace

thread 'rustc' panicked at 'Diagnostic { level: Fatal, message: [("Failed to parse the name of a test block, got err: Diagnostic { level: Fatal, message: [(\"expected identifier, found `#`\", NoStyle)], code: None, span: MultiSpan { primary_spans: [Span { lo: BytePos(96), hi: BytePos(97), ctxt: #0 }], span_labels: [] }, children: [], suggestions: [] }", NoStyle)], code: None, span: MultiSpan { primary_spans: [Span { lo: BytePos(96), hi: BytePos(97), ctxt: #0 }], span_labels: [] }, children: [], suggestions: [] }', /home/austin/.cargo/registry/src/github.com-1ecc6299db9ec823/stainless-0.1.11/src/parse.rs:133:20
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at /checkout/src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:60
             at /checkout/src/libstd/panicking.rs:380
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:390
   4: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:611
   5: std::panicking::begin_panic_new
             at /checkout/src/libstd/panicking.rs:553
   6: std::panicking::begin_panic_fmt
             at /checkout/src/libstd/panicking.rs:521
   7: <stainless::describe::DescribeState as stainless::parse::Parse<(syntax_pos::Span, &'a mut syntax::ext::base::ExtCtxt<'b>, core::option::Option<syntax_pos::symbol::Ident>)>>::parse
             at /home/austin/.cargo/registry/src/github.com-1ecc6299db9ec823/stainless-0.1.11/src/parse.rs:133
   8: stainless::describe::describe
             at /home/austin/.cargo/registry/src/github.com-1ecc6299db9ec823/stainless-0.1.11/src/describe.rs:68
   9: core::ops::function::Fn::call
             at /checkout/src/libcore/ops/function.rs:48
  10: <F as syntax::ext::base::IdentMacroExpander>::expand
             at /checkout/src/libsyntax/ext/base.rs:260
  11: syntax::ext::expand::MacroExpander::expand_invoc
  12: syntax::ext::expand::MacroExpander::expand
  13: syntax::ext::expand::MacroExpander::expand_crate
  14: rustc_driver::driver::phase_2_configure_and_expand::{{closure}}
  15: rustc_driver::driver::phase_2_configure_and_expand
  16: rustc_driver::driver::compile_input
  17: rustc_driver::run_compiler

error: Could not compile `rustc-panic`.

Caused by:
  process didn't exit successfully: `rustc --crate-name rustc_panic src/lib.rs --emit=dep-info,link -C debuginfo=2 --test -C metadata=86f20b59323deae2 -C extra-filename=-86f20b59323deae2 --out-dir /home/austin/scratch/rustc-panic/target/debug/deps -L dependency=/home/austin/scratch/rustc-panic/target/debug/deps --extern stainless=/home/austin/scratch/rustc-panic/target/debug/deps/libstainless-6e9196cf7cd4f8c1.so` (exit code: 101)
```

## Meta

`rustc --version --verbose`:

rustc 1.20.0-nightly (bf0a9e0b4 2017-07-10)
binary: rustc
commit-hash: bf0a9e0b4d3a4dd09717960840798e2933ec7568
commit-date: 2017-07-10
host: x86_64-unknown-linux-gnu
release: 1.20.0-nightly
LLVM version: 4.0

Backtrace: See above

**Comment by sfackler** [comment]
This should be filed against stainless I believe.

**Comment by Mark-Simulacrum** [comment]
Yes, this feels like a stainless bug.
